ambry-cloud/src/test/java/com.github.ambry.cloud/azure/AzureCloudDestinationTest.java

```diff
@@ -210,49 +219,56 @@ public void testExistenceCheck() throws Exception {
-      String proxyPortProperty = scheme + "." + "proxyPort";
-      // Test with no proxy
-      System.clearProperty(proxyHostProperty);
-      System.clearProperty(proxyPortProperty);
-      ConnectionPolicy policy = AzureCloudDestination.getProxyWiseConnectionPolicy(endpoint);
-      assertNull("Expected null proxy", policy.getProxy());
-      // Test with host but no port
-      System.setProperty(proxyHostProperty, proxyHost);
-      policy = AzureCloudDestination.getProxyWiseConnectionPolicy(endpoint);
-      assertNotNull("Expected proxy", policy.getProxy());
-      assertEquals("Wrong host", proxyHost, policy.getProxy().getHostName());
-      assertEquals("Expected default port", -1, policy.getProxy().getPort());
-      // Test with host and port
-      System.setProperty(proxyPortProperty, String.valueOf(proxyPort));
-      policy = AzureCloudDestination.getProxyWiseConnectionPolicy(endpoint);
-      assertNotNull("Expected proxy", policy.getProxy());
-      assertEquals("Wrong host", proxyHost, policy.getProxy().getHostName());
-      assertEquals("Wrong port", proxyPort, policy.getProxy().getPort());
+  public void testAzureConnection() throws Exception {
+    CloudConfig cloudConfig = new CloudConfig(new VerifiableProperties(configProps));
+    AzureCloudConfig azureConfig = new AzureCloudConfig(new VerifiableProperties(configProps));
+    AzureCloudDestination dest = new AzureCloudDestination(cloudConfig, azureConfig, azureMetrics);
+    try {
+      dest.testStorageConnectivity();
+      fail("Expected exception");
+    } catch (IllegalStateException expected) {
     }
+    try {
+      dest.testCosmosConnectivity();
+      fail("Expected exception");
+    } catch (IllegalStateException expected) {
+    }
+  }
+
+  /** Test initializing AzureCloudDestination with a proxy */
+  @Test
+  public void testProxy() throws Exception {
+
+    // Test without proxy
+    CloudConfig cloudConfig = new CloudConfig(new VerifiableProperties(configProps));
+    AzureCloudConfig azureConfig = new AzureCloudConfig(new VerifiableProperties(configProps));
+    AzureCloudDestination dest = new AzureCloudDestination(cloudConfig, azureConfig, azureMetrics);
+    // check operation context proxy
+    assertNull("Expected null proxy in blob op context", dest.getBlobOpContext().getDefaultProxy());
+    assertNull("Expected null proxy in doc client", dest.getDocumentClient().getConnectionPolicy().getProxy());
+
+    // Test with proxy
+    String proxyHost = "azure-proxy.randomcompany.com";
+    int proxyPort = 80;
+    configProps.setProperty(CloudConfig.VCR_PROXY_HOST, proxyHost);
+    configProps.setProperty(CloudConfig.VCR_PROXY_PORT, String.valueOf(proxyPort));
+    cloudConfig = new CloudConfig(new VerifiableProperties(configProps));
+    dest = new AzureCloudDestination(cloudConfig, azureConfig, azureMetrics);
+    assertNotNull("Expected proxy in blob op context", dest.getBlobOpContext().getDefaultProxy());
```

Comment:
Verify host and port of `dest.getBlobOpContext().getDefaultProxy()`?

Reply:
Wanted to but not easy to do.